src/systemd_ctypes/event.py

```diff
@@ -87,6 +92,10 @@ def select(
             self.sd_event.dispatch()
             while self.sd_event.prepare():
                 self.sd_event.dispatch()
+
+        # We can be sure we're not dispatching callbacks anymore
+        libsystemd.Trampoline.deferred = None
```

Comment:
Ah, so that's the final cleanup of the deferred trampolines. Is there only ever one instance of `Selector`? This doesn't feel correct if there could be multiple ones

Reply:
Indeed.  This is why I'm not super happy about this fix.  If we tried to run independent mainloops in separate threads, this would indeed be incorrect.  Trying to do something "more correct" here is hard, though, and we never have anything but the default loop running in the main thread, so ...

Reply:
That's fine. This could do with an assertion that there's only a single instance, but good enough now!